### DISCUSSION TITLE
Toggle macOS's "Do not disturb" or select "Focus mode"

### DISCUSSION THREAD

**Discussion by alexcastillo** [johnlindquist/kit#951] [Q&A]
Hi,

I just discovered this project and I'm very impressed.

I was wondering if there's a way to toggle macOS's "Do not disturb" or select a "Focus" (e.g. personal, work, sleep)?

I'd love to use my Crown brain-computer interface to let my brain state set the macOS/iOS Focus. Or simply toggle "Do not disturb".

Thanks 🙏🏼

**Comment by johnlindquist** [comment]
Hey Alex!

A couple approaches:

1. Create a shortcut and run it from your script. Here's an example: https://github.com/johnlindquist/kit/discussions/676

2. Google for "AppleScript Do Not Disturb", then use the "applescript()" helper:

```
await applescript(`my applescript code`)
```

**Reply by johnlindquist** [reply]
Also, I have a Crown too. Happy to make any examples/specifics you'd like to see :)

**Reply by alexcastillo** [reply]
I can't never turn down a Crown example demo :)

The idea is to automatically switch on "Do not disturb" when someone reaches 70% focus and holds it for 1 minute. The same applies when someone comes down from being focus (30%) and "Do not disturb" should switch off if it was not manually set.

**Comment by alexcastillo** [comment]
Thank you, @johnlindquist!

This is very helpful. I'm leaning towards option 2 as it would require less setup from users.

I found an example for macOS Catalina and it looks like it doesn't work on macOS Monterrey.
[Example](https://apple.stackexchange.com/questions/419207/turn-do-not-disturb-on-or-off-in-an-applescript-app-macos-catalina)

I'll be on the lookout for AppleScripts that may work.

Much appreciated.
